Month3/Project1/Project1/TDinamicArray.h

```diff
@@ -0,0 +1,290 @@
+#pragma once
+#include "Square.h"
+#include <stdlib.h>     
+#include <time.h> 
+
+
+template <typename T>
+class TDinamicArray
+{
+private:
+	int Capacity = 3;
+	int Size = 0;
+	T* Array = new T[Capacity];
+public:
+	TDinamicArray() 
+	{
+	
+	}
+	TDinamicArray(const int EntryCapacity)
+	{
+		Array = new T[EntryCapacity];
+	
+	}
```

Comment:
Careful with initializing the array amount in the private section! What would happen if you call the second constructor here? You would assign array to a _different_ space in memory with size EntryCapacity. What happens to the old one? It would get lost, causing a memory leak. Calling the first constructor is fine, as it doesn't reassign Array.